sympy/core/tests/test_assumptions.py

```diff
@@ -764,32 +767,44 @@ def test_Pow_is_algebraic():
     assert (pi**r).is_algebraic is True
 
 
-def test_Mul_is_prime():
+def test_Mul_is_prime_composite():
     from sympy import Mul
     x = Symbol('x', positive=True, integer=True)
     y = Symbol('y', positive=True, integer=True)
     assert (x*y).is_prime is None
     assert ( (x+1)*(y+1) ).is_prime is False
+    assert ( (x+1)*(y+1) ).is_composite is True
 
     x = Symbol('x', positive=True)
-    assert (x*y).is_prime is None
+    assert ( (x+1)*(y+1) ).is_prime is None
+    assert ( (x+1)*(y+1) ).is_composite is None
 
     assert Mul(6, S.Half, evaluate=False).is_prime is True
```

Comment:
I wonder if the case where any arg is not an integer should simply return None rather than reevaluating the expression. This would be more in keeping with simpler conservative answers rather than trying to do simplification that should be done by the user. My growing feeling is that SymPy tries, too often, to use internal simplification to get a better result but in turn makes things slower because of it. Do you have any thoughts on this?

Reply:
@smichr I agree, I think we can remove evaluating unevaluated Mul/Pow part, since unevaluated Mul/Pow is generated only if explicitly passed 'evaluate=False' by user and it means the user doesn't want simplification nor evaluation of its value.